nox/_option_set.py

```diff
@@ -73,15 +77,51 @@ def __init__(
         self.noxfile = noxfile
         self.merge_func = merge_func
         self.finalizer_func = finalizer_func
-        self.default = default
         self.hidden = hidden
         self.kwargs = kwargs
+        self._default = default
+
+    @property
+    def default(self):
+        if callable(self._default):
+            return self._default()
```

Comment:
You actually have a choice here. You **could** define

```python
def _compute_default(default):
    if callable(default):
        return default()
    return default
```

and then just

```python
self.default = _compute_default(default)
```

in the constructor. However, if `default()` returns a mutable value, this could be dangerous.

Reply:
Yeah I'd rather have it computed every time.